iidm/iidm-api/src/main/java/com/powsybl/iidm/network/limitreduction/computer/AbstractReducedLimitsComputer.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network.limitreduction.computer;
+
+import com.powsybl.iidm.network.LimitType;
+import com.powsybl.iidm.network.ThreeSides;
+import com.powsybl.iidm.network.limitreduction.result.LimitsContainer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * <p>Abstract class responsible for computing reduced limits.</p>
+ * <p>Already computed reduced limits are stored in an internal cache to avoid unnecessary computations.
+ * This cache should be cleared with {@link #clearCache()} when the reductions to apply are changed.</p>
+ *
+ * @author Olivier Perrin {@literal <olivier.perrin at rte-france.com>}
+ */
+public abstract class AbstractReducedLimitsComputer<P, L> implements ReducedLimitsComputer<P, L> {
+    private final Map<CacheKey<P>, LimitsContainer<L>> reducedLimitsCache;
+
+    protected AbstractReducedLimitsComputer() {
+        this.reducedLimitsCache = new HashMap<>();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>Resulting limits are stored in an internal cache, by (identifier's ID, limit type, side), in order to avoid
+     * unnecessary re-computation. If needed, this cache can be cleared using {@link #clearCache()}.</p>
+     *
+     * @param processable The network element for which the reduced limits must be computed
+     * @param limitType The type of the limits to process
+     * @param side The side of <code>processable</code> on which the limits should be retrieved
+     * @return an object containing the original limits and the altered ones
+     */
+    @Override
+    public Optional<LimitsContainer<L>> getLimitsWithAppliedReduction(P processable, LimitType limitType, ThreeSides side) {
```

Comment:
I think (in general) that the simpler is the better. Try to do it simple. Why not just a `getLimits` (because we use a computer, we know that it will be the new ones)?

Reply:
This method was renamed as `computeLimits`.